espflash/src/error.rs

```diff
@@ -271,6 +268,28 @@ pub enum Error {
     AppDescriptorNotPresent(String),
 }
 
+// SlipError doesn't implement `core::error::Error`, so we need to wrap it.
+#[cfg(feature = "serialport")]
+#[derive(Debug)]
+struct SlipErrorWrapper(pub SlipError);
+
+#[cfg(feature = "serialport")]
+impl fmt::Display for SlipErrorWrapper {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "SLIP error: {:?}", self.0)
+    }
+}
+
+#[cfg(feature = "serialport")]
+impl core::error::Error for SlipErrorWrapper {}
+
+#[cfg(feature = "serialport")]
+impl From<SlipError> for Error {
+    fn from(err: SlipError) -> Self {
+        Self::Connection(Box::new(SlipErrorWrapper(err)))
```

Comment:
You already have a `impl From<SlipError> for ConnectionError {` implementation in this file, you don't need a wrapper type.